Lindhart.Analyser.MissingAwaitWarning/Lindhart.Analyser.MissingAwaitWarning/Lindhart.Analyser.MissingAwaitWarning/LindhartAnalyserMissingAwaitWarningAnalyzer.cs

```diff
@@ -10,62 +10,83 @@
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
 
-		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+        private static readonly Type[] AwaitableTypes = new[]
+        {
+            typeof(Task),
+            typeof(Task<>),
+            typeof(ConfiguredTaskAwaitable),
+            typeof(ConfiguredTaskAwaitable<>),
+            typeof(ValueTask),
+            typeof(ValueTask<>),
+            typeof(ConfiguredValueTaskAwaitable),
+            typeof(ConfiguredValueTaskAwaitable<>)
+        };
 
-		public override void Initialize(AnalysisContext context)
-		{
-			context.RegisterSyntaxNodeAction(AnalyseSymbolNode, SyntaxKind.InvocationExpression);
-		}
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
-		private void AnalyseSymbolNode(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext)
-		{
-			if (syntaxNodeAnalysisContext.Node is InvocationExpressionSyntax node)
-			{
-				if (syntaxNodeAnalysisContext
-						.SemanticModel
-						.GetSymbolInfo(node.Expression, syntaxNodeAnalysisContext.CancellationToken)
-						.Symbol is IMethodSymbol methodSymbol)
-				{
-					if (node.Parent is ExpressionStatementSyntax)
-					{
-						// Only checks for the two most common awaitable types. In principle this should instead check all types that are awaitable
-						if (EqualsType(methodSymbol.ReturnType, typeof(Task), typeof(ConfiguredTaskAwaitable)))
-						{
-							var diagnostic = Diagnostic.Create(Rule, node.GetLocation(), methodSymbol.ToDisplayString());
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyseSymbolNode, SyntaxKind.InvocationExpression);
+        }
 
-							syntaxNodeAnalysisContext.ReportDiagnostic(diagnostic);
-						}
-					}
-				}
-			}
-		}
+        private void AnalyseSymbolNode(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext)
+        {
+            if (syntaxNodeAnalysisContext.Node is InvocationExpressionSyntax node)
+            {
+                var symbolInfo = syntaxNodeAnalysisContext
+                    .SemanticModel
+                    .GetSymbolInfo(node.Expression, syntaxNodeAnalysisContext.CancellationToken);
+                
+                if ((symbolInfo.Symbol ?? symbolInfo.CandidateSymbols.FirstOrDefault())
+                    is IMethodSymbol methodSymbol)
+                {
+                    if (node.Parent is ExpressionStatementSyntax)
+                    {
+                        // Only checks for the two most common awaitable types. In principle this should instead check all types that are awaitable
```

Comment:
This comment isn't true anymore - it now checks quite a number of awaitables - in my opinion the comment should just be deleted

Reply:
Oh yeah, I'll remove it.